engine/collection/ingest/engine.go

```diff
@@ -154,23 +154,19 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 		Hex("ref_block_id", tx.ReferenceBlockID[:]).
 		Logger()
 
-	// TODO log the reference block and final height for debug purposes
-	{
-		final, err := e.state.Final().Head()
-		if err != nil {
-			return fmt.Errorf("could not get final height: %w", err)
-		}
-		log = log.With().Uint64("final_height", final.Height).Logger()
-		ref, err := e.state.AtBlockID(tx.ReferenceBlockID).Head()
-		if err == nil {
-			log = log.With().Uint64("ref_block_height", ref.Height).Logger()
-		}
-	}
-
 	log.Info().Msg("transaction message received")
 
-	// using the transaction's reference block, determine which cluster we're in
-	refEpoch := e.state.AtBlockID(tx.ReferenceBlockID).Epochs().Current()
+	// get the state snapshot w.r.t. the reference block
+	refSnapshot := e.state.AtBlockID(tx.ReferenceBlockID)
+	// fail fast if this is an unknown reference
```

Comment:
This is functionally the same as before, but this makes the flow and errors a bit clearer that we are failing because we don't know the reference block, as opposed to failing because we can't find an epoch (which is ultimately because we don't know the reference block)